rp-binary-info/src/types.rs

```diff
@@ -189,6 +194,62 @@ impl IntegerEntry {
     }
 }
 
+/// An entry which contains a pin_encoding.
+#[repr(C)]
+pub struct PinsWithFunctionEntry {
+    header: EntryCommon,
+    pin_encoding: u32,
+}
+
+impl PinsWithFunctionEntry {
+    /// Create a new `PinsWithFunctionEntry`
+    pub const fn new(pin_0: u32, pin_1: u32, func: u32) -> PinsWithFunctionEntry {
+        PinsWithFunctionEntry {
+            header: EntryCommon {
+                data_type: DataType::PinsWithFunction,
+                tag: TAG_RASPBERRY_PI,
+            },
+            pin_encoding: BI_PINS_ENCODING_MULTI
```

Comment:
I think it's worth noting that this is a value with multiple fields, and raspberry pi seem to indicate that there are no more pins by setting two consecutive fields to the same value.

You could also consider using bitbybit::bitfield to describe this 32-bit value, and avoid needing to do a bunch of shifts using magic numbers. I [have examples you can refer to](https://github.com/rust-embedded/cortex-ar/blob/f1d8ad3c08e1917c1f2f2d08aa0072a787c96d0a/cortex-ar/src/register/dfsr.rs#L36)

Reply:
You could then change this to take a slice of pins, rather than only two pins.